services/core/EmailerAgent/emailer/agent.py

```diff
@@ -54,166 +54,315 @@
+            _log.errro('Email messsage body was null, not sending email')
+            return
+
+        self.send_email(from_address, to_addresses, subject, msg)
+
+    def _send_email(self, from_address, to_addresses, mime_message):
+        """
+        The method that actually sends the data to the smtp server to be
+        sent out.
+
+        This method will also publish to the record/sent_email topic so that
+        the email action will be recorded.  The full message content will bw
+        written to the message bus.  The following format is used
+
+        .. code-block:: json
+
+            {
+                "from_address": from_address,
+                "recipients": to_addresses,
+                "subject": mime_message['Subject'],
+                "message_content": mime_message.as_string()
+            }
+
+        :param from_address: The sender of the message
+        :param to_addresses: A list of recipient email addresses.
+        :param mime_message: A `email.mime.text.MimeText` message to be sent.
+        """
+
+        send_successful = False
+        sent_email_record = None
         try:
-            s = smtplib.SMTP(self._smtp)
+            _log.info("Sending email {}".format(mime_message['Subject']))
+            sent_email_record = {"from_address": from_address,
+                                 "recipients": to_addresses,
+                                 "subject": mime_message['Subject'],
+                                 "message_content": mime_message.as_string()}
+            cfg = self.vip.config.get("config")
+            smtp_address = cfg['smtp_address']
+            s = smtplib.SMTP(smtp_address)
+            s.sendmail(from_address, to_addresses, mime_message.as_string())
             s.quit()
-        except socket.gaierror:
-            raise ValueError('Invalid smtp-address')
-	_log.debug('Allow frequency in seconds is: {}'.format(self._allow_frequency_seconds))
-        self._sent_emails = None
-        self._read_store()
-
-    def timestamp(self):
-        return time.mktime(datetime.datetime.now().timetuple())
-
-    def _read_store(self):
-        if os.path.exists('email.store'):
-            with open('email.store', 'r') as f:
-                self._sent_emails = jsonapi.loads(f.read())
-        else:
-            self._sent_emails = {}
-
-    def _write_store(self):
-        with open('email.store', 'w') as f:
-            f.write(jsonapi.dumps(self._sent_emails))
-
-    @PubSub.subscribe('pubsub',
-                      prefix=topics.ALERTS.format(agent_class='',
-                      agent_uuid=''))
-    def onmessage(self, peer, sender, bus, topic, headers, message):
-        _log.info('Alert message for topic: {}'.format(topic))
-        _log.debug("Sending mail for message: {}".format(message))
+            send_successful = True
+        except Exception as e:
+            _log.error(
+                'Unable to send email message: %s' % mime_message.as_string())
```

Comment:
Update status for failure to send email so it is visible in the VC UI. Obviously, sending an alert would be pointless if you can't send an email.